src/test/java/org/sigmah/server/inject/GuiceJUnitRunner.java

```diff
@@ -171,4 +214,72 @@ private static Injector createInjectorFor(final List<Class<?>> classes) throws I
 	private void startPersistUnit() {
 		injector.getInstance(PersistService.class).start();
 	}
+	
+	/**
+	 * Parse the <code>META-INF/context.xml</code> file and try to connect to 
+	 * the database.
+	 * 
+	 * @return <code>true</code> if the connection was successful, 
+	 * <code>false</code> otherwise.
+	 */
+	private boolean isDatabaseAvailable() {
+		
+		boolean available = false;
+		final SAXParserFactory factory = SAXParserFactory.newInstance();
+
+		try {
+			factory.setValidating(false);
+			factory.setFeature("http://xml.org/sax/features/validation", false);
+			factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+			factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+			factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+			factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+
+			final SAXParser parser = factory.newSAXParser();
+			
+			final String[] properties = new String[3];
+			
+			parser.parse(getClass().getResourceAsStream("/META-INF/persistence.xml"), new DefaultHandler() {
+				
+				@Override
+				public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+					if ("property".equals(qName)) {
+						final String name = attributes.getValue("name");
+						final String value = attributes.getValue("value");
+						
+						if (name != null) switch (name) {
+						case "hibernate.connection.driver_class":
+							try {
+								Class.forName(value);
+							} catch (ClassNotFoundException ex) {
+								throw new SAXException("SQL driver '" + value + "' could not be loaded.", ex);
+							}
+							break;
+						case "hibernate.connection.url":
+							properties[0] = value;
+							break;
+						case "hibernate.connection.username":
+							properties[1] = value;
+							break;
+						case "hibernate.connection.password":
+							properties[2] = value;
+							break;
+						default:
+							break;
+						}
+					}
+				}
+				
+			});
+			
+			final Connection testConnection = DriverManager.getConnection(properties[0], properties[1], properties[2]);
+			testConnection.close();
+			
+			available = true;
+		} catch (ParserConfigurationException | SAXException | IOException | SQLException ex) {
+			// Ignored.
```

Comment:
You can uninitialize it.

It will lead to compilation issues because you can't use it (in the `return` statement) without initialize it. That's why I suggest to initialize it to `true` in the `try` clause, to `false` in the `catch SQLException` clause and by throwing an Exception in other cases (the remaining multi-catch clause). By doing that, all the paths that leads to the return statement are with an initialized available. See?